### ISSUE TITLE
Encoding Images with Alpha Channel?

### ISSUE THREAD

**Issue by DavidGetter1** [SunzeY/AlphaCLIP#23]
Hello, my Dataset is completely composed of Images with an alpha channel, my question is if encoding those would already do it in a fashion where only the non-alpha region (the point of interest) is being regarded?
Currently im using CLIP but the results are bad when working with an alpha channel.

**Comment by SunzeY** [comment]
Can you show more details of your case? Original CLIP can only input RGB 3 channel only...

**Comment by DavidGetter1** [comment]
Exactly, so I just disregard the alpha channel and have my background black which is not working good.
```python
        image_array = np.array(image.image)
        image_array = image_array[:, :, :3] #Here I disregard the alpha channel
        image_pil = Image.fromarray(image_array)
        image_pil = image_pil.resize((224, 224), Image.LANCZOS)
        images_to_sprite_data.append(image_pil)
        image_tensor = preprocess_image(image_pil)
        with torch.no_grad():
            image_embedding = model.encode_image(image_tensor)
            embeddings.append(image_embedding.cpu().numpy()[0]) 
```

**Comment by SunzeY** [comment]
We cannot assure 100% attention focus, but you can give it a try with our model, or quickly test our demo!
![image](https://github.com/SunzeY/AlphaCLIP/assets/76475229/1843d2f3-f449-4104-8312-d1f9381d0235)
![image](https://github.com/SunzeY/AlphaCLIP/assets/76475229/08ab0785-a475-4d0c-8c75-aef0c42ecc6f)

**Comment by DavidGetter1** [comment]
I've just created an embedding by following the readme.
```python
image_features = model.visual(image, alpha)
```
I believe the attention is now set by the mask?
I will need to create masks for my dataset and will see whether the results are what im searching for.

**Comment by SunzeY** [comment]
Yes, if you follow README, you can add region focus by mask, which original CLIP is not capable of. You can try this on your own task to see whether it fit your requirement.

**Comment by DavidGetter1** [comment]
Hey I just tried the Model on my use-case and it got me very interesting results. Thank you!
